WordPress/gutenberg
# Patterns: Rename Library to Patterns

Fixes: https://github.com/WordPress/gutenberg/issues/52042

## What?

Renames the Library in the site editor to patterns.

## Why?

- Discovery: this is an opportunity to make patterns front and center as we are introducing the ability to save custom patterns. They should be front and center in the "Design" tab.
- Clarity: library can be obscure and overlaps with other terminology (like Media Library). We didn't get to add font library management in this round, but it would have made things more confusing (would a user expect to see their font library under "Styles" or an item called "Library"?). Calling it patterns removes that ambiguity.
- Presence: patterns is a unique name that has been established in the WP branding parlance for a bit and deserves more clear placement.

## How?

- Rename all paths, strings, and components away from Library to Patterns
- Cleans up the template parts code from the previously shared Templates sidebar navigation screen
- Updates e2e that looked for Library button in sidebar

## Testing Instructions

1. Navigate to the Site Editor, the sidebar should contain a `Patterns` item instead of `Library`
2. Click on Patterns, this page should now have no reference to Library and its path be `/patterns`
3. Confirm you can navigate between all categories of My Patterns, Template Parts, and theme patterns
4. Check you can still manage template parts and custom patterns
5. Test the manage all pages, ensuring that you can navigate back to the Patterns page as expected
6. Make sure the Site Editor > Templates navigation still works

## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/60436221/48c96ad2-452b-4bcf-a05f-0a1afd0b2ce5
